tests/test_demo.py

```diff
@@ -1,8 +1,40 @@
 from datagovharvester import __version__
 from datagovharvester.example import hello
+from utils.json_utilities import open_json
+from pathlib import Path
+import jsonschema
+import pytest
+
+TESTS_DIR = Path(__file__).parents[0]
+
 
 def test_version():
     assert __version__ == "0.1.0"
 
+
 def test_hello():
-    assert hello('name') == "Hello name!"
\ No newline at end of file
+    assert hello('name') == "Hello name!"
+
+
+@pytest.mark.parametrize(
+    "schema, dataset, is_valid",
+    open_json(TESTS_DIR / "fixtures" / "parametrizers" /
+              "params.json")["test_validate_dcat_us"]
+)
```

Comment:
This is really cool and fancy! But maybe overkill, and may make it difficult to read in the future.
I would have built a "parsing" function that abstracts all the "getting json fixture" code. Then I would have a separate test for each json use case, and it would be some copy/paste. In general it would look like the following, with slight customizations for each json file. This allows for comments, and explicit naming for each test case.
```
def test_numerical_title()
    json_to_parse = custom_fixture_function("numerical-title.data.json")
    try:
        jsonschema.validate(json_data, schema=dataset_schema)
        assert false #this should fail validation, numerical titles aren't allowed
    except jsonschema.ValidationError as e:
        error_message = "error: " + e.message + ". offending element: " + e.json_path
        print(error_message)
        assert true
```